securedrop_client/gui/widgets.py

```diff
@@ -1807,6 +1803,208 @@ def _on_left_click(self):
+        buttons_layout.addWidget(usb_cancel_button)
+        buttons_layout.addWidget(retry_export_button)
+        usb_form_layout.addWidget(self.usb_error_message)
+        usb_form_layout.addWidget(usb_instructions)
+        usb_form_layout.addWidget(buttons, alignment=Qt.AlignRight)
+
+        # Passphrase Form
+        self.passphrase_form = QWidget()
+        self.passphrase_form.setObjectName('passphrase_form')
+        passphrase_form_layout = QVBoxLayout()
+        self.passphrase_form.setLayout(passphrase_form_layout)
+        self.passphrase_error_message = SecureQLabel(_(
+            'The passphrase provided did not work. Please try again.'))
+        self.passphrase_error_message.setWordWrap(True)
+        passphrase_instructions = SecureQLabel(_('Enter password for safe USB drive.'))
+        passphrase_instructions.setWordWrap(True)
+        passphrase_label = SecureQLabel(_('Passphrase'))
+        passphrase_label.setObjectName('passphrase_label')
+        self.passphrase_field = QLineEdit()
+        self.passphrase_field.setEchoMode(QLineEdit.Password)
+        buttons = QWidget()
+        buttons_layout = QHBoxLayout()
+        buttons.setLayout(buttons_layout)
+        passphrase_cancel_button = QPushButton(_('CANCEL'))
+        unlock_disk_button = QPushButton(_('SUBMIT'))
+        buttons_layout.addWidget(passphrase_cancel_button)
+        buttons_layout.addWidget(unlock_disk_button)
+        passphrase_form_layout.addWidget(self.passphrase_error_message)
+        passphrase_form_layout.addWidget(passphrase_instructions)
+        passphrase_form_layout.addWidget(passphrase_label)
+        passphrase_form_layout.addWidget(self.passphrase_field)
+        passphrase_form_layout.addWidget(buttons, alignment=Qt.AlignRight)
+        self.passphrase_error_message.hide()
+
+        layout.addWidget(self.starting_export_message)
+        layout.addWidget(self.generic_error)
+        layout.addWidget(self.insert_usb_form)
+        layout.addWidget(self.passphrase_form)
+
+        self.generic_error.hide()
+        self.insert_usb_form.hide()
+        self.passphrase_form.hide()
+
+        usb_cancel_button.clicked.connect(self.close)
+        passphrase_cancel_button.clicked.connect(self.close)
+        retry_export_button.clicked.connect(self._on_retry_export_button_clicked)
+        unlock_disk_button.clicked.connect(self._on_unlock_disk_clicked)
+
+        self._export()
+
+    def _export(self):
+        try:
+            self.controller.run_export_preflight_checks()
+            self._request_passphrase()
+        except ExportError as e:
+            # The first time we see a CALLED_PROCESS_ERROR, tell the user to insert the USB device
+            # in case the issue is that the Export VM cannot start due to a USB device being
+            # unavailable for attachment. According to the Qubes docs:
+            #
+            #   "If the device is unavailable (physically missing or sourceVM not started), booting
```

Comment:
also an edge case (documenting here for posterity and not necessary to include in the documentation) is the case where the machine can't start for other reasons as well (maybe memory pressure or other unknown error)